### PR TITLE
deprecate nuget restore option in ms/vsbuild tasks in favor of nuget installer task

### PR DESCRIPTION
**Pull request by jmyersmsft**

### CODE REVIEW

File: Tasks/MSBuild/Helpers.ps1 [hunk 1]
```diff
@@ -60,6 +60,11 @@ function Invoke-BuildTools {
         [switch]$NoTimelineLogger
     )
 
+    if ($NuGetRestore)
```
Comment (ericsciple) [inline Tasks/MSBuild/Helpers.ps1]: I see you print the warning, but don't short-circuit. I'm guessing that is intentional.

File: Tasks/MSBuild/Helpers.ps1 [hunk 2]
```diff
@@ -60,6 +60,11 @@ function Invoke-BuildTools {
...
+    {
+        Write-Warning (Get-LocalizedString -Key "The `"Restore NuGet Packages`" option is deprecated. To restore NuGet packages in your build, add a NuGet Installer task to your build definition.")
```
Comment (ericsciple) [inline Tasks/MSBuild/Helpers.ps1]: I'm curious, why this change? Is this to avoid updating code in multiple places? If so I've added a mechanism recently for common code to be shared across tasks.

If you make this change, then you need to check to make sure the template corresponds with the change here.

File: Tasks/MSBuild/Helpers.ps1 (hunk 1 above)
Comment (jmyersmsft) [inline Tasks/MSBuild/Helpers.ps1]: Our intention was to print the warning and continue with the existing behavior for now, yes.

File: Tasks/MSBuild/Helpers.ps1 (hunk 2 above)
Comment (jmyersmsft) [inline Tasks/MSBuild/Helpers.ps1]: A checkbox is unfortunately insufficient to work with VSTS-hosted NuGet feeds. We need to know the package source URIs ahead of time in order to configure authentication for them, and I don't think we want to grow *these* tasks into full-fledged NuGet restore tasks

Can you elaborate on "make sure the template corresponds with the change here."?

I don't think the _PS3 version existed when I started this work, so I didn't see it, sorry

Comment (vtbassmatt) [inline Tasks/MSBuild/Helpers.ps1]: @ericsciple it would be awesome to simply not have the checkbox in the new task, and set up the default Visual Studio Build template to include a NuGet Installer step. Is that possible?

Comment (chrisrpatterson) [inline Tasks/MSBuild/Helpers.ps1]: Updating the template should be part of this overall effort.

Sent from Outlook Mobile<https://aka.ms/blhgte>



On Tue, Jan 26, 2016 at 3:37 PM -0800, "Matt Cooper" <notifications@github.com<mailto:notifications@github.com>> wrote:


In Tasks/MSBuild/Helpers.ps1<https://github.com/Microsoft/vso-agent-tasks/pull/1135#discussion_r50919945>:

> @@ -60,6 +60,11 @@ function Invoke-BuildTools {
>          [switch]$NoTimelineLogger
>      )
>
> +    if ($NuGetRestore)
> +    {
> +        Write-Warning (Get-LocalizedString -Key "The `"Restore NuGet Packages`" option is deprecated. To restore NuGet packages in your build, add a NuGet Installer task to your build definition.")


@ericsciple<https://github.com/ericsciple> it would be awesome to simply not have the checkbox in the new task, and set up the default Visual Studio Build template to include a NuGet Installer step. Is that possible?

-

Comment (ericsciple) [inline Tasks/MSBuild/Helpers.ps1]: Templates are here: /Tfs/Service/Build2/Server/Templates

Comment (jmyersmsft) [inline Tasks/MSBuild/Helpers.ps1]: @ericsciple I've sent you an internal PR to update the templates.

**Comment by vtbassmatt** [comment]
@emanuelquintero did you take a look at this one?

**Comment by emanuelquintero** [comment]
Looks good

**Comment by StingyJack** [comment]
Why didnt the checkbox get removed, or at least a note explaining why its being deprecated so we know what is trying to be avoided?
